### PR TITLE
Add terraform for front end pipeline

### PR DESCRIPTION
**Pull request by ColinRTaylor**
This adds the Terraform that will build the front end pipeline in AWS.
Components:

- s3 bucket for static asset hosting
- CloudFront distribution in front of the s3 bucket
- Automatic test, build, and deploy on merge of pull request in to the master branch via CodePipeline & CodeBuild
- AWS policy setup for permissioning

This forces HTTPS on all static assets.

The instructions are in the README and need to be complete by a developer with access to this Github repository and the correct AWS account.

### CODE REVIEW

File: terraform/modules/website/bucket.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+resource "aws_s3_bucket" "bucket_site" {
+
+    bucket = "${var.app_name}-${var.account_id}-${var.git_repository_branch}"
+    acl    = "public-read"
+    force_destroy = true
+    
+    cors_rule {
+        allowed_headers = ["*"]
+        allowed_methods = ["PUT", "POST", "GET", "DELETE"]
+        allowed_origins = ["*"]
```
Comment (jstreb) [inline terraform/modules/website/bucket.tf:10]: This makes sense for now, but we will probably want to lock this down next week.

File: terraform/modules/website/cloudfront.tf [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+        custom_origin_config {
+            http_port = 80
+            https_port = 443
+            origin_protocol_policy = "http-only"
+            origin_ssl_protocols = ["TLSv1"]
+        }
+    }
+
+  default_cache_behavior {
```
Comment (richpressler) [inline terraform/modules/website/cloudfront.tf:24]: Nit: I'm not familiar with this file type, is it normal to have both 4 and 2 space indents in the same file?

Comment (ColinRTaylor) [reply terraform/modules/website/cloudfront.tf:24]: Yep, Terraform has a built in file formatter with predefined rules, so I just used that: https://www.terraform.io/docs/commands/fmt.html

Comment (afraser) [reply terraform/modules/website/cloudfront.tf:24]: 🤯

**Pull request merged**